asteroidsyncserviced/openweathermapparser.h

```diff
@@ -0,0 +1,23 @@
+#ifndef OPENWEATHERMAPPARSER_H
+#define OPENWEATHERMAPPARSER_H
+
+#include <QObject>
+#include <QJsonObject>
+#include <QList>
+#include <QJsonArray>
+#include <QDateTime>
+
+class OpenWeatherMapParser : public QObject
```

Comment:
Since the methods are all stateless you don't really need a class here (an object of type OpenWeatherMapParser doesn't carry any ownership/lifetime semantic)
Or you could do the parsing once in the constructor (because all methods take the same arguments and do the same loop over and over again) and then store the results of the parsing in private members that you could return from getWeatherId(), getTempMin() etc...